arm-containerregistry/2016-06-27-preview/swagger/containerregistry.json

```diff
@@ -0,0 +1,642 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2016-06-27-preview",
+    "title": "ContainerRegistryManagementClient"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ContainerRegistry/checkNameAvailability": {
+      "post": {
+        "tags": [
+          "Operation"
+        ],
+        "description": "Checks whether the container registry name is available.",
+        "operationId": "Operation_CheckNameAvailability",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "registryNameCheckRequest",
+            "in": "body",
+            "description": "The request to check whether the container registry name is available.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/RegistryNameCheckRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/RegistryNameStatus"
+            }
+          }
+        },
+        "deprecated": false
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroup}/providers/Microsoft.ContainerRegistry/registries/{registryName}": {
+      "get": {
+        "tags": [
+          "Registries"
+        ],
+        "description": "Returns the properties for the specified container registry.",
+        "operationId": "Registries_GetProperties",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "resourceGroup",
+            "in": "path",
+            "description": "The resource group name.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "registryName",
+            "in": "path",
+            "description": "The container registry name.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/Registry"
+            }
+          }
+        },
+        "deprecated": false
+      },
+      "put": {
+        "tags": [
+          "Registries"
+        ],
+        "description": "Creates a new container registry with the specified parameters.",
+        "operationId": "Registries_Create",

```

Comment:
Right, our RP will try to do an update if the resource already exists. The issue of making it CreateOrUpdate is that, create and update use different parameters (although technically update uses a subset of parameters from create). In our case, for example, location is a required property in create, but is forbidden to change in update. If the user happens to PUT the correct set of parameters for an existing resource, the RP will return success; otherwise, the RP still returns failure. What is the recommended pattern here?